### PR TITLE
End-to-end test for open-source automation of horizontal resharding.

### PR DESCRIPTION
**Pull request by michael-berlin**
@alainjobart @aaijazi 

I recommend to review the different commits using the "commits" view from Github and not the "files" view because I tried to keep logical changes between the same commit.

### CODE REVIEW

File: go/vt/worker/vtworkerclient/interface.go [hunk 1]
```diff
@@ -12,20 +12,18 @@ import (
...
 
-// ErrFunc is returned by streaming queries to get the error
-type ErrFunc func() error
-
 // VtworkerClient defines the interface used to send remote vtworker commands
 type VtworkerClient interface {
 	// ExecuteVtworkerCommand will execute the command remotely.
 	// NOTE: ErrFunc should only be checked after the returned channel was closed to avoid races.
-	ExecuteVtworkerCommand(ctx context.Context, args []string) (<-chan *logutil.LoggerEvent, ErrFunc)
+	ExecuteVtworkerCommand(ctx context.Context, args []string) (<-chan *logutil.LoggerEvent, vtctlclient.ErrFunc)
```
Comment (alainjobart) [inline go/vt/worker/vtworkerclient/interface.go]: I would keep using the local ErrFunc here (so just revert this file changes). That eliminates a dependency to vtctlclient, and keeps both API definitions self-contained.

**Comment by alainjobart** [comment]
LGTM good stuff!

File: go/vt/worker/vtworkerclient/interface.go (hunk 1 above)
Comment (michael-berlin) [inline go/vt/worker/vtworkerclient/interface.go]: DONE

**Comment by michael-berlin** [comment]
@alainjobart 

PTAL

**Comment by alainjobart** [comment]
LGTM

**Comment by michael-berlin** [comment]
@alainjobart 

Can you PTAL? I had to add two more commits because I broke the worker.py test. After that I'm ready to merge.

**Comment by alainjobart** [comment]
LGTM

**Pull request merged**